sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRegistryAsyncClient.java

```diff
@@ -0,0 +1,186 @@
+    private static final String CONTINUATIONLINKHEADERNAME = "Link";
+    private static final String LINKMATCHER = "(<(.+)>;.*)";
+    private static final String ORDERBYQUERYPARAM = "&orderby=";
+
+    /**
+     * Initializes an instance of ContainerRegistries client.
+     *
+     * @param serviceClient the service client implementation.
+     */
+    ContainerRegistryAsyncClient(ContainerRegistriesImpl serviceClient) {
+        this.serviceClient = serviceClient;
+    }
+
+    /**
+     * List repositories.
+     *
+     * @return list of repositories.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws AcrErrorsException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedFlux<String> listRepositories() {
+        ListRepositoriesOptions options = new ListRepositoriesOptions();
+        return this.listRepositories(options);
+    }
+
+    /**
+     * List repositories.
+     *
+     * @param options for the list repositories call.
+     * @return list of repositories.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws NullPointerException thrown if the parameter is null.
+     * @throws AcrErrorsException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedFlux<String> listRepositories(ListRepositoriesOptions options) {
+        if (options == null) {
+            throw logger.logExceptionAsError(new NullPointerException("options can't be null"));
+        }
+
+        Integer pageSize = options.getMaxPageSize();
+        return new PagedFlux<>(
+            () -> withContext(context -> listRepositoriesSinglePageAsync(pageSize, context)),
+            token -> withContext(context -> listRepositoriesNextSinglePageAsync(token, context)));
+    }
+
+    private PagedResponse<String> getPagedResponseWithContinuationToken(PagedResponse<String> listRepositoriesResponse) {
+        String continuationLink = null;
+        HttpHeaders headers = listRepositoriesResponse.getHeaders();
+
+        if (headers != null) {
+            String continuationLinkHeader = headers.getValue(CONTINUATIONLINKHEADERNAME);
+            if (!CoreUtils.isNullOrEmpty(continuationLinkHeader)) {
+                // TODO: ACR follows docker API spec which has continuation token in header.
+                // It currently returns it in the header without the url which they intend to fix
+                // in the upcoming releases.
+                Pattern linkPattern = Pattern.compile(LINKMATCHER);
+                Matcher matcher = linkPattern.matcher(continuationLinkHeader);
+                if (matcher.matches()) {
```

Comment:
Does the entire `continuationLinkHeader` need to be contained in the pattern or only a portion?

For example given the pattern `hello`, which of the following is allowed to pass this conditional:

1. `hello world`
2. `hello`

If `matches` is used only #2 will return true, if `find` is used both will return true.

Reply:
In this particular case yes. The entire response is in that format.

---
In reply to: [602629307](https://github.com/Azure/azure-sdk-for-java/pull/20139#discussion_r602629307) [](ancestors = 602629307)